templates/add-persistence-attribute-PASSWORD_RESET_POST_CHALLENGE/code.js

```diff
@@ -33,43 +39,82 @@ exports.onExecutePostChallenge = async (event, api) => {
         return api.access.deny('missing client id');
     }
 
-    const clientId = event.secrets.CLIENT_ID;
-
     if (!event.secrets.CLIENT_SECRET) {
         return api.access.deny('missing client secret');
     }
 
-    const clientSecret = event.secrets.CLIENT_SECRET;
-
     if (!event.secrets.TENANT_DOMAIN) {
         return api.access.deny('missing tenant domain');
     }
 
-    const domain = event.secrets.TENANT_DOMAIN;
+    // if event.secrets.AUDIENCE is set, make use of the parameter for validating JWT is being used by intended Audience
+    if (event.secrets.AUDIENCE) {
+        authenticationClientOptions.audience = event.secrets.AUDIENCE;
```

Comment:
is it just me or is `authenticationClientOptions` not defined before it is used?